gnat2goto/driver/tree_walk.adb

```diff
@@ -1377,12 +1350,12 @@ package body Tree_Walk is
    function Do_And_Then (N : Node_Id) return Irep is
       L : constant Node_Id := Left_Opnd (N);
       R : constant Node_Id := Right_Opnd (N);
-      Expr : constant Irep := New_Irep (I_Op_And);
+      Expr : constant Irep := Make_Op_And
+        (I_Type => CProver_Bool_T,
```

Comment:
Also, that has nothing to do with your PR, but I do find it weird that our constructors for binary operators do not get the operands and we have to add them through `Append_Op`. At least it's surprising given that the other constructors do get their operands in their interface.

Reply:
As per in-person talk: Our `And` and `Or` operators are actually n-ary operators, hence they take a list of operands rather than having them explicitly passed (I don't particularly like this either, I'm considering adding an overload to `goto-utils` that takes an array of operands in cases where we know which ones we want).